File: Kotlin/src/Board.kt
import java.util.Set

class Board(val blackCells : Set<Position>) {

    fun isBlack(position : Position) : Boolean {
        return blackCells.contains(position);
    }

    fun flip(position: Position) {
        if(blackCells.contains(position))
            blackCells.remove(position)
        else
            blackCells.add(position)
    }

}

File: Dart/src/Board.dart
class Board {
  
  Set<Cell> _blackCells;
  Iterable<Cell> get blackCells() => _blackCells;
  
  Board(Iterable<Cell> initialBlackCells) {
    _blackCells = new Set.from(initialBlackCells);
  }
  
  bool isBlack(Cell cell) => _blackCells.contains(cell);
  
  void flipColor(Cell cell) {
    if(_blackCells.contains(cell))
      _blackCells.remove(cell);
    else
      _blackCells.add(cell);
  }
}
